src/services/index.js

```
@@ -0,0 +1,34 @@
+import { get, set, cloneDeepWith } from 'lodash';
+import { bind, getContainer } from '@globality/nodule-config';
+import getServiceWrappers from './wrapper';
+
+export function cloneClients(obj) {
+    return cloneDeepWith(obj, node => (
+        typeof node === 'function' ?
+            async (req, args) => node(req, args) :
+            Object.keys(node).reduce(
+                (acc, key) => ({ ...acc, [key]: cloneClients(node[key]) }),
+                {},
+            )
+    ));
+}
+
+function optimizeServices(services) {
```

I'd love to understand what is optimized here. It's not obvious why `get/set` helps us.

---

I think the get/set is to protect against specifying a request name in the config that wasn't defined for the client but I guess we want to fail loudly if we do that so I'll remove it.